config/value.go

```diff
@@ -333,10 +333,41 @@ func derefType(t reflect.Type) reflect.Type {
 	return t
 }
 
+func convertPrimitiveValue(value interface{}, targetType reflect.Type) (interface{}, error) {
+	ret, err := convertVal(value, targetType)
+	if ret == nil {
```

Comment:
The err check here is not relevant here. The only concern is to check if return type is nil (as in convertible primitive type or not) Thats why we check `v != nil && err == nil` in Try* method and perform further type casting. I can check err here, but it will just be a no op, since Try* methods are taking care of the rest.